src/domains/Session/SessionService.php

```diff
@@ -72,21 +76,20 @@ public function currentUser()
 
     public function createPasswordHash($sPassword)
     {
-        // init
-        $encryptedPassword = (object) array(
-            'salt' => bin2hex(random_bytes(32)),
-            'iterations' => 50000
-        );
+        // hash
+        $sHashedPassword = password_hash($sPassword, self::PASSWORD_HASH_ALGORITM, self::PASSWORD_HASH_OPTIONS);
 
-        // encrypt
-        $encryptedPassword->hash = hash_pbkdf2('sha256', $sPassword, $encryptedPassword->salt, $encryptedPassword->iterations, 64);
+        // validate
+        if (empty($sHashedPassword)) {
```

Comment:
Check for explicit false:

https://secure.php.net/manual/en/function.password-hash.php

Reply:
I explicitly didn't, as it is a fail-safe. If something incorrect comes out you'll want to catch it.

Reply:
I get your point, but I don't think there is much merit to it. According to the interface it either returns a hashed string or it returns false. Unless you want to guard yourself against failures of the PHP Core (but then I'd regex the whole hash instead of just checking empty), this will never match anything else than false.

Now it makes it look like it can actually return an empty string, which is confusing imho.

Reply:
Ha. I did this based on experience. Didn't know where from, until I found the compatibility package https://github.com/ircmaxell/password_compat again:

> It is very important that you should check the return value of `password_hash` prior to storing it, because `false` or `null` may be returned if it encountered an error.

I remember an earlier version also told to test against empty string or even 60 chars length. I got used to an empty check to prevent the worst.

Anyway, this also mainly depends on the coding standards in Mimoto. Whether `empty()` is to be avoided.